[crashpad] Build error on x64-linux

Package: crashpad:x64-linux@2022-09-05#4

**Host Environment**

- Host: x64-linux
- Compiler: GNU 11.4.0
-    vcpkg-tool version: 2024-03-14-7d353e869753e5609a1f1a057df3db8fd356e49d
    vcpkg-scripts version: ad3bae574 2024-04-12 (10 days ago)

**To Reproduce**

`vcpkg install `
**Failure logs**

```
-- Fetching https://chromium.googlesource.com/crashpad/crashpad 261679b3d2f3336d8531ed38e110254c3e2d1c10...
-- Extracting source /usr/local/share/vcpkg/downloads/crashpad-261679b3d2f3336d8531ed38e110254c3e2d1c10.tar.gz
-- Using source at /usr/local/share/vcpkg/buildtrees/crashpad/src/4c3e2d1c10-1c4f0a1e28.clean
-- Fetching https://chromium.googlesource.com/chromium/mini_chromium 5654edb4225bcad13901155c819febb5748e502b...
-- Extracting source /usr/local/share/vcpkg/downloads/crashpad-5654edb4225bcad13901155c819febb5748e502b.tar.gz
-- Using source at /usr/local/share/vcpkg/buildtrees/crashpad/src/b5748e502b-44507d16bb.clean
-- Fetching https://chromium.googlesource.com/linux-syscall-support 9719c1e1e676814c456b55f5f070eabad6709d31...
-- Extracting source /usr/local/share/vcpkg/downloads/crashpad-9719c1e1e676814c456b55f5f070eabad6709d31.tar.gz
-- Using source at /usr/local/share/vcpkg/buildtrees/crashpad/src/bad6709d31-ab881aeb4b.clean
-- Generating build (x64-linux-dbg)...
-- Generating build (x64-linux-rel)...
-- Building (x64-linux-dbg)...
CMake Error at scripts/cmake/vcpkg_execute_build_process.cmake:134 (message):
    Command failed: /usr/local/share/vcpkg/downloads/tools/ninja/1.10.2-linux/ninja -C /usr/local/share/vcpkg/buildtrees/crashpad/x64-linux-dbg client client:common util third_party/mini_chromium/mini_chromium/base handler:crashpad_handler
    Working Directory: /usr/local/share/vcpkg/buildtrees/crashpad/src/4c3e2d1c10-1c4f0a1e28.clean
    See logs for more information:
      /usr/local/share/vcpkg/buildtrees/crashpad/build-x64-linux-dbg-out.log

Call Stack (most recent call first):
  scripts/cmake/vcpkg_build_ninja.cmake:3 (vcpkg_execute_build_process)
  scripts/cmake/vcpkg_build_ninja.cmake:24 (z_vcpkg_build_ninja_build)
  /home/runner/work/Cpp-Examples/Cpp-Examples/build/vcpkg_installed/x64-linux/share/vcpkg-gn/vcpkg_gn_install.cmake:94 (vcpkg_build_ninja)
  buildtrees/versioning_/versions/crashpad/63c757af964d6d69f6abdeb0a3849889ad3531a9/portfile.cmake:109 (vcpkg_gn_install)
  scripts/ports.cmake:175 (include)



```
<details><summary>/usr/local/share/vcpkg/buildtrees/crashpad/build-x64-linux-dbg-out.log</summary>

```
ninja: Entering directory `/usr/local/share/vcpkg/buildtrees/crashpad/x64-linux-dbg'
[1/210] STAMP obj/build/default_exe_manifest_win.stamp
[2/210] STAMP obj/third_party/lss/lss.stamp
[3/210] CXX obj/third_party/mini_chromium/mini_chromium/base/files/base.file_path.o
FAILED: obj/third_party/mini_chromium/mini_chromium/base/files/base.file_path.o 
clang++ -MMD -MF obj/third_party/mini_chromium/mini_chromium/base/files/base.file_path.o.d -D_FILE_OFFSET_BITS=64 -I../src/4c3e2d1c10-1c4f0a1e28.clean/third_party/mini_chromium/mini_chromium -Igen  -Wall -Wendif-labels -Werror -Wextra -Wextra-semi -Wheader-hygiene -Wnewline-eof -Wno-missing-field-initializers -Wno-unused-parameter -Wsign-compare -Wstring-conversion -Wvla -fno-exceptions -fno-rtti -fno-strict-aliasing -fobjc-call-cxx-cdtors -fstack-protector-all -fvisibility-inlines-hidden -fvisibility=hidden -fPIC -pthread -m64 -g -Wexit-time-destructors -Wimplicit-fallthrough -std=c++17 -c ../src/4c3e2d1c10-1c4f0a1e28.clean/third_party/mini_chromium/mini_chromium/base/files/file_path.cc -o obj/third_party/mini_chromium/mini_chromium/base/files/base.file_path.o
In file included from ../src/4c3e2d1c10-1c4f0a1e28.clean/third_party/mini_chromium/mini_chromium/base/files/file_path.cc:10:
../src/4c3e2d1c10-1c4f0a1e28.clean/third_party/mini_chromium/mini_chromium/base/logging.h:23:28: error: unknown type name 'uint32_t'
using LoggingDestination = uint32_t;
                           ^
../src/4c3e2d1c10-1c4f0a1e28.clean/third_party/mini_chromium/mini_chromium/base/logging.h:29:8: error: unknown type name 'LoggingDestination'
enum : LoggingDestination {
       ^
../src/4c3e2d1c10-1c4f0a1e28.clean/third_party/mini_chromium/mini_chromium/base/logging.h:47:3: error: unknown type name 'LoggingDestination'
  LoggingDestination logging_dest = LOG_DEFAULT;
  ^
3 errors generated.
[4/210] CXX obj/third_party/mini_chromium/mini_chromium/base/debug/base.alias.o
[5/210] CXX obj/third_party/mini_chromium/mini_chromium/base/base.logging.o
FAILED: obj/third_party/mini_chromium/mini_chromium/base/base.logging.o 
clang++ -MMD -MF obj/third_party/mini_chromium/mini_chromium/base/base.logging.o.d -D_FILE_OFFSET_BITS=64 -I../src/4c3e2d1c10-1c4f0a1e28.clean/third_party/mini_chromium/mini_chromium -Igen  -Wall -Wendif-labels -Werror -Wextra -Wextra-semi -Wheader-hygiene -Wnewline-eof -Wno-missing-field-initializers -Wno-unused-parameter -Wsign-compare -Wstring-conversion -Wvla -fno-exceptions -fno-rtti -fno-strict-aliasing -fobjc-call-cxx-cdtors -fstack-protector-all -fvisibility-inlines-hidden -fvisibility=hidden -fPIC -pthread -m64 -g -Wexit-time-destructors -Wimplicit-fallthrough -std=c++17 -c ../src/4c3e2d1c10-1c4f0a1e28.clean/third_party/mini_chromium/mini_chromium/base/logging.cc -o obj/third_party/mini_chromium/mini_chromium/base/base.logging.o
In file included from ../src/4c3e2d1c10-1c4f0a1e28.clean/third_party/mini_chromium/mini_chromium/base/logging.cc:5:
../src/4c3e2d1c10-1c4f0a1e28.clean/third_party/mini_chromium/mini_chromium/base/logging.h:23:28: error: unknown type name 'uint32_t'
using LoggingDestination = uint32_t;
                           ^
../src/4c3e2d1c10-1c4f0a1e28.clean/third_party/mini_chromium/mini_chromium/base/logging.h:29:8: error: unknown type name 'LoggingDestination'
enum : LoggingDestination {
       ^
../src/4c3e2d1c10-1c4f0a1e28.clean/third_party/mini_chromium/mini_chromium/base/logging.h:47:3: error: unknown type name 'LoggingDestination'
  LoggingDestination logging_dest = LOG_DEFAULT;
  ^
../src/4c3e2d1c10-1c4f0a1e28.clean/third_party/mini_chromium/mini_chromium/base/logging.cc:76:1: error: unknown type name 'LoggingDestination'
LoggingDestination g_logging_destination = LOG_DEFAULT;
^
4 errors generated.
[6/210] CXX obj/third_party/mini_chromium/mini_chromium/base/files/base.scoped_file.o
[7/210] CXX obj/compat/linux/sys/compat.mman_memfd_create.o
[8/210] CXX obj/tools/tool_support.tool_support.o
ninja: build stopped: subcommand failed.
```
</details>

**Additional context**

<details><summary>vcpkg.json</summary>

```
{
  "$schema": "https://raw.githubusercontent.com/microsoft/vcpkg-tool/main/docs/vcpkg.schema.json",
  "name": "cpp-examples",
  "version": "0.0.1",
  "description": "manifest",
  "dependencies": [
    "glog",
    "breakpad",
    "benchmark",
    "gtest",
    {
      "name": "openssl",
      "features": [
        "tools"
      ]
    },
    {
      "name": "curl",
      "features": [
        "openssl",
        "brotli",
        "c-ares",
        "http2",
        "tool"
      ]
    },
    "crashpad"
  ],
  "builtin-baseline": "ad3bae57455a3c3ce528fcd47d8e8027d0498add"
}

```
</details>

Reproduce on `x64-linux`:
```
FAILED: obj/third_party/mini_chromium/mini_chromium/base/synchronization/base.lock_impl_posix.o
clang++ -MMD -MF obj/third_party/mini_chromium/mini_chromium/base/synchronization/base.lock_impl_posix.o.d -D_FILE_OFFSET_BITS=64 -I../src/4c3e2d1c10-1c4f0a1e28.clean/third_party/mini_chromium/mini_chromium -Igen  -Wall -Wendif-labels -Werror -Wextra -Wextra-semi -Wheader-hygiene -Wnewline-eof -Wno-missing-field-initializers -Wno-unused-parameter -Wsign-compare -Wstring-conversion -Wvla -fno-exceptions -fno-rtti -fno-strict-aliasing -fobjc-call-cxx-cdtors -fstack-protector-all -fvisibility-inlines-hidden -fvisibility=hidden -fPIC -pthread -m64 -g -Wexit-time-destructors -Wimplicit-fallthrough -std=c++17 -c ../src/4c3e2d1c10-1c4f0a1e28.clean/third_party/mini_chromium/mini_chromium/base/synchronization/lock_impl_posix.cc -o obj/third_party/mini_chromium/mini_chromium/base/synchronization/base.lock_impl_posix.o
In file included from ../src/4c3e2d1c10-1c4f0a1e28.clean/third_party/mini_chromium/mini_chromium/base/synchronization/lock_impl_posix.cc:9:
../src/4c3e2d1c10-1c4f0a1e28.clean/third_party/mini_chromium/mini_chromium/base/logging.h:23:28: error: unknown type name 'uint32_t'
using LoggingDestination = uint32_t;
                           ^
../src/4c3e2d1c10-1c4f0a1e28.clean/third_party/mini_chromium/mini_chromium/base/logging.h:29:8: error: unknown type name 'LoggingDestination'
enum : LoggingDestination {
       ^
../src/4c3e2d1c10-1c4f0a1e28.clean/third_party/mini_chromium/mini_chromium/base/logging.h:47:3: error: unknown type name 'LoggingDestination'
  LoggingDestination logging_dest = LOG_DEFAULT;
  ^
3 errors generated.

```

--- a/ports/crashpad/fix-stdint.patch
+++ b/ports/crashpad/fix-stdint.patch
@@ -1,0 +1,26 @@
+diff --git a/base/logging.h b/base/logging.h
+index a123535..a87de3d 100644
+--- a/base/logging.h
++++ b/base/logging.h
+@@ -7,7 +7,7 @@
+ 
+ #include <assert.h>
+ #include <errno.h>
+-
++#include <stdint.h>
+ #include <limits>
+ #include <sstream>
+ #include <string>
+diff --git a/base/strings/utf_string_conversion_utils.h b/base/strings/utf_string_conversion_utils.h
+index ed99a23..0ba3fea 100644
+--- a/base/strings/utf_string_conversion_utils.h
++++ b/base/strings/utf_string_conversion_utils.h
+@@ -6,7 +6,7 @@
+ #define MINI_CHROMIUM_BASE_STRINGS_UTF_STRING_CONVERSION_UTILS_H_
+ 
+ #include <string>
+-
++#include <stdint.h>
+ namespace base {
+ 
+ inline bool IsValidCodepoint(uint32_t code_point) {

--- a/ports/crashpad/portfile.cmake
+++ b/ports/crashpad/portfile.cmake
@@ -18,17 +18,21 @@ function(checkout_in_path PATH URL REF)
         OUT_SOURCE_PATH DEP_SOURCE_PATH
         URL "${URL}"
         REF "${REF}"
+        PATCHES
+          "${PATCH_FILE}"
     )
     file(RENAME "${DEP_SOURCE_PATH}" "${PATH}")
     file(REMOVE_RECURSE "${DEP_SOURCE_PATH}")
 endfunction()
 
+set(PATCH_FILE "fix-stdint.patch")
 # mini_chromium contains the toolchains and build configuration
 checkout_in_path(
     "${SOURCE_PATH}/third_party/mini_chromium/mini_chromium"
     "https://chromium.googlesource.com/chromium/mini_chromium"
     "5654edb4225bcad13901155c819febb5748e502b"
 )
+set(PATCH_FILE "")
 
 if(VCPKG_TARGET_IS_LINUX)
     # fetch lss

--- a/ports/crashpad/portfile.cmake
+++ b/ports/crashpad/portfile.cmake
@@ -25,8 +25,8 @@ function(checkout_in_path PATH URL REF)
     file(REMOVE_RECURSE "${DEP_SOURCE_PATH}")
 endfunction()
 
-set(PATCH_FILE "fix-stdint.patch")
 # mini_chromium contains the toolchains and build configuration
+set(PATCH_FILE "fix-stdint.patch")
 checkout_in_path(
     "${SOURCE_PATH}/third_party/mini_chromium/mini_chromium"
     "https://chromium.googlesource.com/chromium/mini_chromium"

--- a/ports/crashpad/vcpkg.json
+++ b/ports/crashpad/vcpkg.json
@@ -1,7 +1,7 @@
 {
   "name": "crashpad",
   "version-date": "2022-09-05",
-  "port-version": 4,
+  "port-version": 5,
   "description": [
     "Crashpad is a crash-reporting system.",
     "Crashpad is a library for capturing, storing and transmitting postmortem crash reports from a client to an upstream collection server. Crashpad aims to make it possible for clients to capture process state at the time of crash with the best possible fidelity and coverage, with the minimum of fuss."
@@ -10,6 +10,7 @@
   "license": "Apache-2.0",
   "supports": "linux | osx | (windows & !uwp)",
   "dependencies": [
+    "curl",
     {
       "name": "vcpkg-cmake-get-vars",
       "host": true

--- a/versions/baseline.json
+++ b/versions/baseline.json
@@ -1986,7 +1986,7 @@
     },
     "crashpad": {
       "baseline": "2022-09-05",
-      "port-version": 4
+      "port-version": 5
     },
     "crashrpt": {
       "baseline": "1.4.3",

--- a/versions/c-/crashpad.json
+++ b/versions/c-/crashpad.json
@@ -1,6 +1,11 @@
 {
   "versions": [
     {
+      "git-tree": "f7161b19eece4ad2643d6b2baafdb5fd6ec57572",
+      "version-date": "2022-09-05",
+      "port-version": 5
+    },
+    {
       "git-tree": "63c757af964d6d69f6abdeb0a3849889ad3531a9",
       "version-date": "2022-09-05",
       "port-version": 4

